backend/src/core/annotation.py

```diff
@@ -105,25 +105,15 @@ def process_tasks(annotation_queue, genomic_unit_collection):  # pylint: disable
                         annotation_value = genomic_unit_collection.find_genomic_unit_annotation_value(
                             annotation_unit.genomic_unit, missing
                         )
-                        if annotation_value:
-                            annotation_unit.genomic_unit[missing] = annotation_value
-                        else:
-                            ready = False
+                        ready = annotation_unit.ready_for_annotation(annotation_value, missing)
 
                 if not ready and not latest:
-                    annotation_unit.increment_delay_count()
+                    missing_dependencies, logger_message = annotation_unit.should_continue_annotation()
 
-                    if annotation_unit.should_continue_annotation():
-                        logger.info(
-                            '%s Delaying Annotation, Missing Dependency...', format_annotation_logging(annotation_unit)
-                        )
+                    if not missing_dependencies:
```

Comment:
This reads as "If it is not missing dependencies, the put it back onto the queue".  I am pretty sure you mean the following case "if it is not missing dependencies but is not the latest either, put it back on the queue, right?"

I know this is a refactor for in progress work, so maybe some additional documentation on what your future plans for handling the case of 'isn't missing dependencies but isn't latest'.

Reply:
Line 110 already checks if the annotation unit is not ready and not latest. `latest` is a variable that will indicate if the annotated unit has been annotated previously in another case, and that annotation is latest/most up to date. Right now, `latest` is just set to True, since the versioning code hasn't gone in yet.

Line 113, is checking if this annotation unit has any missing_dependencies at all. Since there are some annotation configurations that don't have/need dependencies. 

I can add comments for this in the `process_tasks` function for now, until the versioning code is added. And try to make line 113 more readable.

We can also meet to talk about this.

Reply:
We followed up and talked about this! Its good to go.